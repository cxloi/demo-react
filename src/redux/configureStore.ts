import {
  // applyMiddleware,
  // compose,
  createStore,
} from 'redux';

// import monitorReducersEnhancer from "./enhancers/monitorReducers";
// import loggerMiddleware from "./middleware/logger";
import { rootReducer } from './reducers';

export default function configureStore() {
  // const middlewares = [loggerMiddleware, thunkMiddleware]
  // const middlewareEnhancer = applyMiddleware(...middlewares)

  // const enhancers = [middlewareEnhancer, monitorReducersEnhancer]
  // const composedEnhancers = compose(...enhancers)

  const store = createStore(
    rootReducer,
    // preloadedState,
    // composedEnhancers
  );

  return store;
}
