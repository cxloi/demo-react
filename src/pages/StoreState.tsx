import React from 'react';
import { useSelector } from 'react-redux';
import { selectWholeStoreState } from 'redux/selectors';
import {
  makeStyles,
  createStyles,
  Theme as AugmentedTheme,
  useTheme,
} from '@material-ui/core/styles';

const useStyles = makeStyles(
  (theme: AugmentedTheme) =>
    createStyles({
      root: {
        ...theme.typography.h4,
        color: theme.status.danger,
        margin: theme.spacing(2),
        [theme.breakpoints.up('sm')]: {
          margin: theme.spacing(4),
        },
        '&$checked': {
          // color: theme.status.danger,
        },
      },
      sub: {
        // color: theme.colors.ctaBackground,
      },
      checked: {},
    }),
  // eslint-disable-next-line function-paren-newline
);

const StoreState: React.FC = () => {
  const wholeStoreState = useSelector(selectWholeStoreState);
  const classes = useStyles();
  const themeObj = useTheme();

  return (
    <div data-testid="storeStatePage">
      <h2 className={classes.root}>{`Store State (${themeObj.name})`}</h2>
      <div className={classes.sub} style={{ whiteSpace: 'pre-wrap' }}>
        {JSON.stringify(wholeStoreState, null, 4)}
      </div>
    </div>
  );
};

export default StoreState;
