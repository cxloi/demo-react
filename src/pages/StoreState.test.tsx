import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import StoreState from './StoreState';

const mockState = {
  count: 1,
  user: {
    isLoggedIn: true,
    userDetails: {
      userName: 'Dan Abramov',
      age: 38,
    },
  },
};
const mockStore = configureStore([])(mockState);

const componentWithStore = <Provider store={mockStore}><StoreState /></Provider>;
const renderComponent = () => render(componentWithStore);

describe('StoreState', () => {
  it('should render the whole store', () => {
    const { getByTestId } = renderComponent();
    const textElement = getByTestId('storeStatePage').textContent;
    expect(textElement).toContain('"count": 1,');
    expect(textElement).toContain('"userName": "Dan Abramov"');
    expect(textElement).toContain('"age": 38');
  });

  test('render snapshot', () => {
    const tree = renderer
      .create(componentWithStore)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
