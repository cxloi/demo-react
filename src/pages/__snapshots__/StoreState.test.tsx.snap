// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreState render snapshot 1`] = `
<div
  data-testid="storeStatePage"
>
  <h2
    className="makeStyles-root-4"
  >
    Store State (undefined)
  </h2>
  <div
    className="makeStyles-sub-5"
    style={
      Object {
        "whiteSpace": "pre-wrap",
      }
    }
  >
    {
    "count": 1,
    "user": {
        "isLoggedIn": true,
        "userDetails": {
            "userName": "Dan Abramov",
            "age": 38
        }
    }
}
  </div>
</div>
`;
